pkg/service/controller.go

```diff
@@ -271,7 +271,16 @@ func (c *ControllerService) ControllerPublishVolume(
 
 	err = c.virtClient.AddVolumeToVM(c.infraClusterNamespace, vmName, addVolumeOptions)
 	if err != nil {
-		klog.Errorf("Failed adding volume %s to VM %s, %v", dvName, vmName, err)
+		// Need to ignore already exists error due to multiple add requests
+		if !strings.Contains(err.Error(), fmt.Sprintf("Unable to add volume [%s] because it already exists", dvName)) {
+			klog.Errorf("failed adding volume %s to VM %s, %v", dvName, vmName, err)
+			return nil, err
+		}
+	}
+
+	err = c.virtClient.EnsureVolumeAvailable(c.infraClusterNamespace, vmName, dvName)
+	if err != nil {
+		klog.Errorf("volume %s failed to be ready in time in VM %s, %v", dvName, vmName, err)
```

Comment:
what happens if this timeout occurs? Will the publish be reattempted? I just want to make sure this is eventually consistent

Reply:
Yes, the publish will be re-attempted, which will get to this spot again and we will wait again. If there is a real problem hotplugging, then it will continue failing here. But I wanted to give the user some feedback in between attempts.